style_guides/architecture.md

```diff
@@ -34,3 +34,36 @@ plugin_root:
   <dd>This folder is where code that is useful on both the client and the server belongs. A consistent example of this is constants, but this could apply to helper modules as well.</dd>
   <dd><strong>NOTE</strong>: If you'd like to avoid adding <code>../common</code> to your public code, you could use <em>webpackShims</em> to resolve the path without traversing backwards.</dd>
 </dl>
+
+### Subdirectories
+
+As a plugin gets more complex, it becomes important to organize your code into subdirectories. Each subdirectory should contain an `index.js` file that exposes the contents of that directory.
```

Comment:
I don't think so, but if this document is not broadly worded enough to encompass all of the code in Kibana, then IMHO the doc as a whole should be fixed.